assets/js/src/site/plugins/pum-cookies.js

```diff
@@ -1,183 +1,211 @@
-                cookies = settings.cookies || [],
-                cookie = null,
-                i;
-
-            if (cookies.length) {
-                for (i = 0; cookies.length > i; i += 1) {
-                    cookie = cookies[i];
-                    $popup.popmake('addCookie', cookie.event, cookie.settings);
-                }
-            }
-        });
-
-}(jQuery, document));
+(function($, document, undefined) {
+	"use strict";
+
+	var setCookie = function(settings) {
+		$.pm_cookie(
+			settings.name,
+			true,
+			settings.session ? null : settings.time,
+			settings.path ? pum_vars.home_url || "/" : null
+		);
+		pum.hooks.doAction("popmake.setCookie", settings);
+	};
+
+	$.extend($.fn.popmake.methods, {
+		addCookie: function(type) {
+			// Method calling logic
+
+			pum.hooks.doAction("popmake.addCookie", arguments);
+
+			if ($.fn.popmake.cookies[type]) {
+				return $.fn.popmake.cookies[type].apply(
+					this,
+					Array.prototype.slice.call(arguments, 1)
+				);
+			}
+			if (window.console) {
+				console.warn("Cookie type " + type + " does not exist.");
+			}
+			return this;
+		},
+		setCookie: setCookie,
+		checkCookies: function(settings) {
+			var i,
+				ret = false;
+
+			if (
+				settings.cookie_name === undefined ||
+				settings.cookie_name === null ||
+				settings.cookie_name === ""
+			) {
+				return false;
+			}
+
+			switch (typeof settings.cookie_name) {
+				case "object":
+				case "array":
+					for (i = 0; settings.cookie_name.length > i; i += 1) {
+						if (
+							$.pm_cookie(settings.cookie_name[i]) !== undefined
+						) {
+							ret = true;
+						}
+					}
+					break;
+				case "string":
+					if ($.pm_cookie(settings.cookie_name) !== undefined) {
+						ret = true;
+					}
+					break;
+			}
+
+			pum.hooks.doAction("popmake.checkCookies", settings, ret);
+
+			return ret;
+		}
+	});
+
+	$.fn.popmake.cookies = $.fn.popmake.cookies || {};
+
+	$.extend($.fn.popmake.cookies, {
+		on_popup_open: function(settings) {
+			var $popup = PUM.getPopup(this);
+			$popup.on("pumAfterOpen", function() {
+				$popup.popmake("setCookie", settings);
+			});
+		},
+		on_popup_close: function(settings) {
+			var $popup = PUM.getPopup(this);
+			$popup.on("pumBeforeClose", function() {
+				$popup.popmake("setCookie", settings);
+			});
+		},
+		form_submission: function(settings) {
+			var $popup = PUM.getPopup(this);
+
+			settings = $.extend(
+				{
+					form: "",
+					formInstanceId: "",
+					only_in_popup: false
+				},
+				settings
+			);
+
+			PUM.hooks.addAction("pum.integration.form.success", function(
+				form,
+				args
+			) {
+				if (!settings.form.length) {
+					return;
+				}
+
+				if (
+					PUM.integrations.checkFormKeyMatches(
+						settings.form,
+						settings.formInstanceId,
+						args
+					)
+				) {
+					if (
+						(settings.only_in_popup &&
+							args.popup.length &&
+							args.popup.is($popup)) ||
+						!settings.only_in_popup
+					) {
```

Comment:
Fixed spaces to tabs so it shows the whole file changed. The only actual code changes if this conditional.

Reply:
PHPStorm doesn't allow line by line staging which is ashame, but VSCode does which is nice, you can highlight the changed lines, stage them, commit, then stage the rest of the file for a second commit.